pkg/apis/direct.csi.min.io/v1beta3/types.go

```diff
@@ -0,0 +1,279 @@
+// This file is part of MinIO Direct CSI
+// Copyright (c) 2021 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package v1beta3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	DirectCSIVolumeFinalizerPVProtection    = Group + "/pv-protection"
+	DirectCSIVolumeFinalizerPurgeProtection = Group + "/purge-protection"
+
+	DirectCSIDriveFinalizerDataProtection = Group + "/data-protection"
+	DirectCSIDriveFinalizerPrefix         = Group + ".volume/"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:storageversion
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type DirectCSIDrive struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   DirectCSIDriveSpec   `json:"spec"`
+	Status DirectCSIDriveStatus `json:"status,omitempty"`
+}
+
+type DirectCSIDriveSpec struct {
+	// +optional
+	RequestedFormat *RequestedFormat `json:"requestedFormat,omitempty"`
+	// required
+	DirectCSIOwned bool `json:"directCSIOwned"`
+	// +optional
+	DriveTaint map[string]string `json:"driveTaint,omitempty"`
```

Comment:
Can we remove this field as we don't use it? We can also do this in a separate PR

cc @wlan0